bokehjs/src/lib/models/sources/column_data_source.ts

```diff
@@ -168,19 +167,16 @@ export class ColumnDataSource extends ColumnarDataSource {
   }
 
   attributes_as_json(include_defaults: boolean = true, value_to_json = ColumnDataSource._value_to_json): any {
-    const attrs: Attrs = {}
-    const obj = this.serializable_attributes()
-    for (const key of keys(obj)) {
-      let value = obj[key]
-      if (key === 'data')
-        value = encode_column_data(value as Data, this._shapes)
-
-      if (include_defaults)
-        attrs[key] = value
-      else if (key in this._set_after_defaults)
-        attrs[key] = value
+    const attributes: Attrs = {} // Object.create(null)
+    for (const [name, prop] of this) {
+      if (prop.syncable && (include_defaults || prop.dirty)) {
+        let value = prop.get_value()
+        if (name === "data")
+          value = encode_column_data(value as Data, this._shapes)
```

Comment:
Well this condition is still pretty ugly, suggest an issue or at least a TODO so maybe it can be improved in the future

Reply:
This is actually fixed (replaced with a more generic approach, which doesn't require this special case) in one of my local branches. I will be submitting a PR in near future.